three-nodes-hybrid-deployment.md

```diff
@@ -0,0 +1,130 @@
+
+![QPS with modified config](/media/best-practices/three-nodes-final-config-qps.png)
+
+## Parameter adjustment
+
+Performance jitter occurs in the image above, because the default thread pool configuration and the resource allocation to background tasks are for machines with sufficient resources. In the hybrid deployment scenario, the resources are shared among multiple components, so you need to limit the resource consumption via configuration parameters.
+
+The final cluster configuration for this test is as follows:
+
+{{< copyable "" >}}
+
+```yaml
+tikv:
+    readpool.unified.max-thread-count: 6
+    server.grpc-concurrency: 2
+    storage.scheduler-worker-pool-size: 2
+    gc.max-write-bytes-per-sec: 300K
+    rocksdb.max-background-jobs: 3
+    rocksdb.max-sub-compactions: 1
+    rocksdb.rate-bytes-per-sec: “200M”
+
+  tidb:
+    performance.committer-concurrency: 4
+    performance.max-procs: 8
+```
+
+The following sections introduce the meanings and the adjustment methods of these parameters.
+
+### Configuration of TiKV thread pool size
+
+This section offers best practices for adjusting parameters that relate to the resource allocation of thread pools for foreground applications. Reducing these thread pool sizes will compromise performance, but in the hybrid deployment scenario with limited resources, the cluster itself is hard to achieve high performance. In this scenario, the overall stability of the cluster is preferred over performance.
+
+If you conduct an actual load test, you can first use the default configuration and observe the actual resource usage of each thread pool. Then you can adjust the corresponding configuration items and reduce the sizes of the thread pools that have lower usage.
+
+#### `readpool.unified.max-thread-count`
+
+The default value of this parameter is 80% of the number of machine threads. In a hybrid deployment scenario, you need to manually calculate and specify this value. You can first set it to 80% of the expected number of CPU threads used by TiKV.
+
+#### `server.grpc-concurrency`
+
+This parameter defaults to `4`. Because in the existing deployment plan the CPU resources are limited, the actual requests are few. You can observe the monitoring panel, lower the value of this parameter, and keep the usage rate below 80%.
+
+In this test, the value of this parameter is set to `2`. Observe the **gRPC poll CPU** panel and you can see that the usage rate is just around 80%.
+
+![gRPC Pool CPU](/media/best-practices/three-nodes-grpc-pool-usage.png)
+
+#### `storage.scheduler-worker-pool-size`
+
+When TiKV detects that the CPU core number of the machine is greater than or equal to `16`, this parameter value defaults to `8`. When the CPU core number is smaller than `16`, the parameter value defaults to `4`. This parameter is used when TiKV converts complex transaction requests to simple key-value reads or writes, but the scheduler thread pool does not performs any reads or writes.
```

Comment:
"reads" are not mentioned in [the Chinese version](https://docs.pingcap.com/zh/tidb/stable/three-nodes-hybrid-deployment). Please check.
```suggestion
When TiKV detects that the CPU core number of the machine is greater than or equal to `16`, this parameter value defaults to `8`. When the CPU core number is smaller than `16`, the parameter value defaults to `4`. This parameter is used when TiKV converts complex transaction requests to simple key-value reads or writes, but the scheduler thread pool does not performs any writes.
```